test/list_multimap/built_list_multimap_test.dart

```diff
@@ -0,0 +1,403 @@
+    });
+
+    test('hashes to same value for same contents', () {
+      final multimap1 = new BuiltListMultimap<int, String>(
+          {1: ['1'], 2: ['2', '2'], 3: ['3']});
+      final multimap2 = new BuiltListMultimap<int, String>(
+          {1: ['1'], 2: ['2', '2'], 3: ['3']});
+
+      expect(multimap1.hashCode, multimap2.hashCode);
+    });
+
+    test('hashes to different value for different keys', () {
+      final multimap1 = new BuiltListMultimap<int, String>(
+          {1: ['1'], 2: ['2', '2'], 3: ['3']});
+      final multimap2 = new BuiltListMultimap<int, String>(
+          {1: ['1'], 2: ['2', '2'], 4: ['3']});
+
+      expect(multimap1.hashCode, isNot(multimap2.hashCode));
+    });
+
+    test('hashes to different value for different values', () {
+      final multimap1 = new BuiltListMultimap<int, String>(
+          {1: ['1'], 2: ['2', '2'], 3: ['3']});
+      final multimap2 = new BuiltListMultimap<int, String>(
+          {1: ['1'], 2: ['2', '3'], 3: ['3']});
+
+      expect(multimap1.hashCode, isNot(multimap2.hashCode));
+    });
+
+    test('caches hash', () {
+      final multimap =
+          new BuiltListMultimap<Object, Object>({1: [new _HashcodeOnlyOnce()]});
+
+      multimap.hashCode;
+      multimap.hashCode;
+    });
+
+    test('compares equal to same contents', () {
+      final multimap1 = new BuiltListMultimap<int, String>(
+          {1: ['1'], 2: ['2', '2'], 3: ['3']});
+      final multimap2 = new BuiltListMultimap<int, String>(
+          {1: ['1'], 2: ['2', '2'], 3: ['3']});
+
+      expect(multimap1 == multimap2, isTrue);
+    });
+
+    test('compares not equal to different type', () {
+      expect(
+          new BuiltListMultimap<int, String>({1: ['1'], 2: ['2'], 3: ['3']}) ==
+              '', isFalse);
+    });
+
+    test('compares not equal to different length BuiltListMultimap', () {
+      expect(
+          new BuiltListMultimap<int, String>({1: ['1'], 2: ['2'], 3: ['3']}) ==
+              new BuiltListMultimap<int, String>({1: ['1'], 2: ['2']}),
+          isFalse);
+    });
+
+    test('compares not equal to different hashcode BuiltListMultimap', () {
+      expect(BuiltCollectionTestHelpers.overridenHashcodeBuiltListMultimap(
+              {1: ['1'], 2: ['2'], 3: ['3']}, 0) ==
+          BuiltCollectionTestHelpers.overridenHashcodeBuiltListMultimap(
+              {1: ['1'], 2: ['2'], 3: ['3']}, 1), isFalse);
+    });
+
+    test('compares not equal to different content BuiltListMultimap', () {
+      expect(BuiltCollectionTestHelpers.overridenHashcodeBuiltListMultimap(
+              {1: ['1'], 2: ['2'], 3: ['3']}, 0) ==
+          BuiltCollectionTestHelpers.overridenHashcodeBuiltListMultimap(
+              {1: ['1'], 2: ['2'], 3: ['3', '4']}, 0), isFalse);
+    });
+
+    test('provides toString() for debugging', () {
+      expect(new BuiltListMultimap<int, String>({1: ['1'], 2: ['2'], 3: ['3']})
+          .toString(), '{1: [1], 2: [2], 3: [3]}');
```

Comment:
Seeing this... do we want to include <K,V>{...}? I should probably file an issue for the other collections as well.